process/marshal.go

```diff
@@ -58,25 +58,25 @@ func (w *Process) UnmarshalJSON(b []byte) error {
 
 func (w *Process) unmarshalNode(nodeType string, marshalData []byte) (*Process_Node, error) {
```

Comment:
I suggest to get rid of those custom marshal function and use jsonpb or proto package to encode and decode instead of default json encoder.
The default json encode (encoding/json) is not working well with interface and oneof fields.